### PR TITLE
Patrick json file

### PR DESCRIPTION
**Pull request by pkallas**
Fixes # .

Overview
Adds json file for Patrick to data directory; a weird bug was found where searching by a skill in the search bar returns all learners even if they don't have the skill, but the function is logging that it is returning properly.

Data Model / DB Schema Changes
<N/A if no changes>

Environment / Configuration Changes
<N/A if no changes>

Notes
Do not merge until bug is fixed.

### CODE REVIEW

**Comment by pkallas** [comment]
I refactored filterByName, which fixes the bug. I still don't know the cause of the original bug, so I preserved the code on a separate branch to look at later. But the branch is ready to be merged now!

**Comment by pkallas** [comment]
Cause of the bug was revealed to be how filter behaves. Because an array was being returned for every learner's skills (even if it was empty), filter interpreted the array it received as a truthy, and was thus returning every learner.

File: src/server/data/learners/PatrickKallas.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+			"id": 104,
+			"projects": "Learners Guild Hackathon"
+		},
+		{
+			"id": 105,
+			"projects": "Talent App"
+		}
+	],
+	"alumni": false
+}
```
Comment (judytuna) [inline src/server/data/learners/PatrickKallas.json:47]: is it your plan to do another pull request to just this file, removing the ids?

**Review by judytuna — APPROVED** [review]

File: src/server/data/learners/PatrickKallas.json (hunk 1 above)
Comment (pkallas) [reply src/server/data/learners/PatrickKallas.json:47]: In the pull request Jack makes once he is done with automatically creating ids, we can do a pull request to remove all of the ids at once.
